adapters/minutemedia/minutemedia.go

```diff
@@ -0,0 +1,121 @@
+	return &adapter{
+		endpointURL: config.Endpoint,
+	}, nil
+}
+
+// MakeRequests prepares the HTTP requests which should be made to fetch bids.
+func (a *adapter) MakeRequests(openRTBRequest *openrtb2.BidRequest, _ *adapters.ExtraRequestInfo) (requestsToBidder []*adapters.RequestData, errs []error) {
+	org, err := extractOrg(openRTBRequest)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("extractOrg: %w", err))
+		return nil, errs
+	}
+
+	openRTBRequestJSON, err := json.Marshal(openRTBRequest)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("marshal bidRequest: %w", err))
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+
+	return append(requestsToBidder, &adapters.RequestData{
+		Method:  http.MethodPost,
+		Uri:     a.endpointURL + "?publisher_id=" + org,
+		Body:    openRTBRequestJSON,
+		Headers: headers,
+	}), nil
+}
+
+// MakeBids unpacks the server's response into Bids.
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, _ *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if adapters.IsResponseStatusCodeNoContent(responseData) {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(responseData); err != nil {
+		return nil, []error{err}
+	}
+
+	var response openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	bidResponse.Currency = response.Cur
+
+	var errs []error
+
+	for _, seatBid := range response.SeatBid {
+		for i, bid := range seatBid.Bid {
+			bidType, err := getMediaTypeForBid(bid)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     &seatBid.Bid[i],
+				BidType: bidType,
+			})
+		}
+	}
+
+	return bidResponse, errs
+}
+
+func extractOrg(openRTBRequest *openrtb2.BidRequest) (string, error) {
+	var err error
+	for _, imp := range openRTBRequest.Imp {
+		var bidderExt adapters.ExtImpBidder
+		if err = json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			return "", fmt.Errorf("unmarshal bidderExt: %w", err)
+		}
+
+		var impExt openrtb_ext.ImpExtMinuteMedia
+		if err = json.Unmarshal(bidderExt.Bidder, &impExt); err != nil {
+			return "", fmt.Errorf("unmarshal ImpExtMinuteMedia: %w", err)
+		}
+
+		if impExt.Org != "" {
+			return strings.TrimSpace(impExt.Org), nil
+		}
+	}
+
+	return "", errors.New("no org supplied")
+}
+
+func getMediaTypeForBid(bid openrtb2.Bid) (openrtb_ext.BidType, error) {
+	switch bid.MType {
+	case openrtb2.MarkupBanner:
+		return openrtb_ext.BidTypeBanner, nil
+	case openrtb2.MarkupVideo:
+		return openrtb_ext.BidTypeVideo, nil
+	default:
+		return "", fmt.Errorf("unsupported MType %d", bid.MType)
```

Comment:
Please wrap this error as a bad server response to provide context:

```
&errortypes.BadServerResponse{
  Message: fmt.Sprintf("unsupported MType %d", bid.MType)
}
```

Reply:
Done 👍🏻